### PR TITLE
Add fileset documentation

### PR DESCRIPTION
**Pull request by KutsenkoA**

### CODE REVIEW

File: extra_docs/jfs/README.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  file
+}
+```
+
+### [Upload a file to the fileset](#upload)
+If you already set up a fileset, you are able to upload files. One or multiple files can be uploaded at once. `upload()` 
+returns an observable and you need to call `subscribe()` to start uploading process. Each uploaded file will emit a 
+*fileRecord*, which contains *id* and *status*. of uploaded file.
+```typescript
+jfs.fileset('my_awesome_fileset').upload([
```
Comment (MishaJexia) [inline extra_docs/jfs/README.md]: Can we change from my_awesome_fileset to fileset_name

File: extra_docs/jfs/README.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ The first status you will get in response from `upload()` is `in_progress` (if you did not set true to the 
+`subscribeForTheFileUploading` config property). It means that file record has been created and currently file is 
+uploading to the persistent storage (whether AWS bucket or something else). 
+There are two ways to track further file status changes - you can subscribe to the fileset changes with 
+`realTimeModule` or you can set `subscribeForTheFileUploading` config option and SDK subscribes to that 
+status automatically.
+
+Use `realTimeModule`:
+```typescript
+import { jexiaClient, fileOperations, realTime } from 'jexia-sdk-js';
```
Comment (MishaJexia) [inline extra_docs/jfs/README.md]: Do I need to use jexia-sdk-js/node or jexia-sdk-js/browser ?

File: extra_docs/jfs/README.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    })
+  ))
+).subscribe();
+```
+As you can see, there are some details you should take care of: filtering events by record id (`watch()` returns 
+all events from the fileset), unsubscribing from watch observable with `takeWhile`. Better to use the next option.
+
+Use `subscribeForTheFileUploading` config property (notice that you still need to activate realTimeModule):
+```typescript
+import { jexiaClient, fileOperations, realTime } from 'jexia-sdk-js';
```
Comment (MishaJexia) [inline extra_docs/jfs/README.md]: Same here... /node /browser?

File: extra_docs/jfs/README.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+jexiaClient().init({
+  projectID: 'your-project-id',
+  key: 'your-project-key',
+  secret: 'your-project-secret'
+}, jfs, realTime());
+
+const fileset = jfs.fileset('my_awesome_fileset');
+
+fileset.upload(record).subscribe(fileRecord => {
```
Comment (MishaJexia) [inline extra_docs/jfs/README.md]: would be good to have record example added to this code. All in one.

File: extra_docs/jfs/README.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      name: 'logo.png',
+      size: 42341,
+      type: 'png',
+      url: 'https://...',
+      status: 'completed' }
+  */
+});
+```
+
+### [Fileset CRUD operations](#crud)
```
Comment (MishaJexia) [inline extra_docs/jfs/README.md]: I think we do not have Create operator. Update + RUD?

Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: we have `insert` which is basically CREATE

File: extra_docs/jfs/README.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
+# Jexia Fileset Module (JFS)
+
+JFS is used for the uploading, fetching and manipulating files as any other user data. Basically, fileset has the same
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: _JFS is used for ~~the~~ uploading_

File: extra_docs/jfs/README.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
+# Jexia Fileset Module (JFS)
+
+JFS is used for the uploading, fetching and manipulating files as any other user data. Basically, fileset has the same 
+functionality as dataset, but provides a way to maintain large data objects (files).
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: suggestion: _Basically, ~~fileset has the same functionality as dataset~~ **fileset is a dataset**, but provides..._

File: extra_docs/jfs/README.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
+# Jexia Fileset Module (JFS)
+
+JFS is used for the uploading, fetching and manipulating files as any other user data. Basically, fileset has the same 
+functionality as dataset, but provides a way to maintain large data objects (files).  
+Before uploading any file you need to manually create a fileset with web management application 
+[app.jexia.com](https://app.jexia.com).
+You can also set AWS bucket and credentials so that your files will be automatically uploaded to the Amazon storage.
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: suggestion: _automatically uploaded to ~~the~~ **YOUR** Amazon storage_

File: extra_docs/jfs/README.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+### [Initialize](#init) 
+To use fileset features, just add fileOperations module to the client initialization. Notice, that for Node.js
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: format `fileOperations` as code

File: extra_docs/jfs/README.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+you have to import fileOperations from *jexia-sdk-js/node*, but if you running your application in browser you need to
+use *jexia-sdk-js/browser* path. It's because file format is different under these two platforms.
+```typescript
+import { jexiaClient, fileOperations } from 'jexia-sdk-js/node'; // 'jexia-sdk-js/browser' for browser applications
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: please use double quotes to keep consistency with _most_ of the code (and all other examples)

File: extra_docs/jfs/README.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+     default: false */
+  subscribeForTheFileUploading: boolean
+  /* Timeout for uploading files
+     default: 120000 */
+  uploadTimeout: number
+});
+```
+
+### [Files format](#format)
+Under Node.js `jfs` accepts [read stream](#https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_class_fs_readstream),
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: we should [add `engines` to `package.json`](https://docs.npmjs.com/files/package.json#engines) to match this node version

File: extra_docs/jfs/README.md [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+As you can see, there are some details you should take care of: filtering events by record id (`watch()` returns 
+all events from the fileset), unsubscribing from watch observable with `takeWhile`. Better to use the next option.
+
+Use `subscribeForTheFileUploading` config property (notice that you still need to activate realTimeModule):
+```typescript
+import { jexiaClient, fileOperations, realTime } from 'jexia-sdk-js/node';
+import * as fs from 'fs';
+
+const jfs = fileOperations({
+  subscribeForTheFileUploading: true
```
Comment (marlosin) [inline extra_docs/jfs/README.md]: `subscribeForTheFileUploading` doesn't explain much, what do you think about changing it to `fileCompletedOnly` or `watchFileCompleted` ?

File: extra_docs/jfs/README.md (hunk 11 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: well, as long as our SDK can be used also with browser, I don't think we need to add node as an engine. NPM will require node then, do we want this?

File: extra_docs/jfs/README.md (hunk 6 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: 👍

File: extra_docs/jfs/README.md (hunk 7 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: 👍

File: extra_docs/jfs/README.md (hunk 8 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: 👍

File: extra_docs/jfs/README.md (hunk 9 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: 👍

File: extra_docs/jfs/README.md (hunk 10 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: done

File: extra_docs/jfs/README.md (hunk 12 above)
Comment (KutsenkoA) [reply extra_docs/jfs/README.md]: changed to `uploadWaitForCompleted`. I used `upload` as a prefix to point out that this config option is related to file uploading

File: extra_docs/jfs/README.md (hunk 11 above)
Comment (marlosin) [reply extra_docs/jfs/README.md]: we could add browsers too, but I suggest we discuss that afterwards :+1:

**Pull request merged**